### PR TITLE
Declare `\l_graphics_search_ext_seq` in l3backend in non-LaTeX

### PR DESCRIPTION
**Pull request by muzimuzhi**
This seq was declared in `l3graphics` but is set in a per-backend basis. Since backend driver is always loaded, even without `l3graphics`, declaring it in `l3backend-graphics` avoids a setting-before-declaration issue.

### CODE REVIEW

File: l3kernel/l3debug.dtx [hunk 1]
```diff
@@ -787,6 +787,7 @@
       \prop_set_eq:NN
       \prop_set_from_keyval:Nn
       \seq_set_eq:NN
+      \seq_set_from_clist:Nn
```
Comment (muzimuzhi) [inline l3kernel/l3debug.dtx:790]: Wondering why this change along won't fail the checks.

Comment (muzimuzhi) [reply l3kernel/l3debug.dtx:790]: Oh, the problem of setting `\l_graphics_search_ext_seq` before declaration only appears in non-LaTeX formats:

https://github.com/latex3/latex3/blob/05c22772e096b48aac4addfbb8b3b93a70e803ee/l3backend/l3backend-graphics.dtx#L66-L76

One of the backends
https://github.com/latex3/latex3/blob/05c22772e096b48aac4addfbb8b3b93a70e803ee/l3backend/l3backend-graphics.dtx#L83-L89

Comment (muzimuzhi) [reply l3kernel/l3debug.dtx:790]: And, since generic loader (`expl3-generic.tex`) loads both the expl3 kernel (`expl3-code.tex`) and backend driver, there's currently no way to turn on debugging between loading kernel and driver.

Currently (before this PR) with little tweak `l3graphics.sty` can be loaded in plain TeX:
```tex
\input expl3-generic.tex\relax

\ExplSyntaxOn
\cs_new_eq:NN \ProvidesExplPackage \use_none:nnnn
\input l3graphics.sty\relax
\ExplSyntaxOff

\end
```
Compile it and you'll get an error
```
! LaTeX Error: Control sequence \l_graphics_search_ext_seq already defined.
```

Thus the question becomes: does `l3graphics` aim to be used in non-LaTeX formats (in the future), or we live with the small flaw that a variable would be set before declaration when `l3graphics` is used in unsupported formats? @josephwright

Comment (muzimuzhi) [reply l3kernel/l3debug.dtx:790]: Now `\l_graphics_search_ext_seq` is declared in `l3graphics` in LaTeX format, and in backend drivers otherwise.

**Comment by muzimuzhi** [comment]
Not sure if a new test file loading `l3graphics` in plainTeX is needed.

**Comment by muzimuzhi** [comment]
Not sure if a new test file loading `l3graphics` in plainTeX is needed.

File: l3backend/l3backend-graphics.dtx [hunk 2]
```diff
@@ -61,20 +61,28 @@
 %    \end{macrocode}
 %
 % \begin{macro}{\@@_backend_loaded:n}
+% \begin{variable}[no-user-doc]{\l_graphics_search_ext_seq}
```
Comment (muzimuzhi) [inline l3backend/l3backend-graphics.dtx:64]: This results in an undefined label, because the outer most setting wins.

> End of definition for `\__graphics_backend_loaded:n` and `\l_graphics_search_ext_seq`. This function is documented on page ??.)

I think this is a special use case that need not be sorted in `l3doc`, hence I will push a workaround.

**Comment by josephwright** [comment]
I need to think about this one a bit. The model is that nothing 'general' should be declared in `l3backend` at all: if it's not engine-specific in some way, it belongs in the 'parent' module. But I see the timing issue here, at least at present. (I do have a PR open to move `l3graphics` to `l3kernel`, which would nicely address that.)

**Comment by muzimuzhi** [comment]
> (I do have a PR open to move `l3graphics` to `l3kernel`, which would nicely address that.)

Oh I forgot it (see #1338).

Since moving `l3graphics` to `l3kernel` automatically avoids the issue this PR tries to resolve and before that `l3graphics` is provided only as a LaTeX2e package, not officially loadable in plainTeX, I will close this PR.

**Comment by josephwright** [comment]
@muzimuzhi  I fyou are sure :) I'll rebase the other PR to perhaps remind the rest of the team ;)

**Comment by muzimuzhi** [comment]
> > Since moving `l3graphics` to `l3kernel` automatically avoids the issue this PR tries to resolve
>
> If you are sure :)

See https://github.com/latex3/latex3/pull/1338#discussion_r1504625942. 😄
